app/components/channel_list/categories/body/channel/channel_list_item.tsx

```diff
@@ -92,14 +94,14 @@ const ChannelListItem = ({channel, isActive, isOwnDirectMessage, isMuted, myChan
         };
     });
 
-    const switchChannels = () => {
+    const switchChannels = useCallback(() => {
```

Comment:
We saw some performance degradation on the TouchableOpacity component, and the only apparent possible reason was that this action was not memoized with a useCallback.

Reply:
We should revisit this one as in the end it could've been the `ChannelIcon`